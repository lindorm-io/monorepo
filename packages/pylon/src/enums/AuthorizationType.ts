export enum AuthorizationType {
  Basic = "basic",
  Bearer = "bearer",
  None = "none",
}
