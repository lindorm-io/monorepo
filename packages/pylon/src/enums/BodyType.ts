export enum BodyType {
  Json = "json",
  Multipart = "multipart",
  Text = "text",
  UrlEncoded = "urlencoded",
  Unknown = "unknown",
}
