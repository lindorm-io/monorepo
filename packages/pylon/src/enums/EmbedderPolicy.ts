export enum EmbedderPolicy {
  Credentialless = "credentialless",
  RequireCorp = "require-corp",
  UnsafeNone = "unsafe-none",
}
