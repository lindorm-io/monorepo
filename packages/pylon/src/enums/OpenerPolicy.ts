export enum OpenerPolicy {
  SameOrigin = "same-origin",
  SameOriginAllowPopups = "same-origin-allow-popups",
  UnsafeNone = "unsafe-none",
}
