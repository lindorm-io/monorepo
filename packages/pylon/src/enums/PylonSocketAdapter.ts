export enum PylonSocketAdapter {
  Memory = "memory",
  Redis = "redis",
}
