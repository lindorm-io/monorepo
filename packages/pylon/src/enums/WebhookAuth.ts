export enum WebhookAuth {
  AuthHeaders = "auth_headers",
  Basic = "basic",
  ClientCredentials = "client_credentials",
  None = "none",
}
