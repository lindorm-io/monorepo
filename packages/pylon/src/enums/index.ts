export * from "./AuthorizationType";
export * from "./BodyType";
export * from "./EmbedderPolicy";
export * from "./OpenerPolicy";
export * from "./PylonSocketAdapter";
