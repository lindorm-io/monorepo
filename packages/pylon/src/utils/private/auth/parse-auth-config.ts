import { PkceMethod } from "@lindorm/enums";
import { merge } from "@lindorm/utils";
import { PylonAuthConfig, PylonAuthOptions } from "../../../types";

const DEFAULT: Omit<PylonAuthConfig, "clientId" | "clientSecret" | "issuer"> = {
  codeChallengeMethod: PkceMethod.S256,
  errorRedirect: "/error",
  pathPrefix: "/auth",
  tokenExpiry: "1d",

  defaults: {
    acrValues: null,
    audience: null,
    maxAge: null,
    prompt: null,
    responseType: "code",
    scope: ["openid", "offline_access", "email", "profile"],
  },

  dynamicRedirectDomains: [],

  expose: {
    accessToken: false,
    idToken: false,
    scope: false,
    subject: false,
  },

  cookies: {
    login: "pylon_login_session",
    logout: "pylon_logout_session",
  },

  refresh: {
    maxAge: "1h",
    mode: "half_life",
  },

  staticRedirect: {
    login: null,
    logout: null,
  },
};

export const parseAuthConfig = (options: PylonAuthOptions): PylonAuthConfig => {
  const merged = merge<PylonAuthConfig>(DEFAULT, options);

  merged.errorRedirect =
    options.errorRedirect ?? merged.pathPrefix + merged.errorRedirect;

  return merged;
};
