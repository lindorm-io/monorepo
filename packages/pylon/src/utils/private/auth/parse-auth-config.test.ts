import { PkceMethod } from "@lindorm/enums";
import { parseAuthConfig } from "./parse-auth-config";

describe("parseAuthConfig", () => {
  test("should merge defaults with minimal options", () => {
    expect(
      parseAuthConfig({
        clientId: "test-client-id",
        clientSecret: "test-client-secret",
        issuer: "https://issuer.com",
      }),
    ).toMatchSnapshot();
  });

  test("should merge defaults with options", () => {
    expect(
      parseAuthConfig({
        clientId: "test-client-id",
        clientSecret: "test-client-secret",
        issuer: "https://issuer.com",

        codeChallengeMethod: PkceMethod.Plain,
        tokenExpiry: "1d",

        dynamicRedirectDomains: ["https://client.com"],

        refresh: {
          maxAge: "6m",
          mode: "none",
        },
        staticRedirect: {
          login: "https://client.com/login/static",
        },
      }),
    ).toMatchSnapshot();
  });
});
