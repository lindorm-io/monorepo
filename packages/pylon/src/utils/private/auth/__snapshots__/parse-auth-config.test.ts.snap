// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseAuthConfig should merge defaults with minimal options 1`] = `
{
  "clientId": "test-client-id",
  "clientSecret": "test-client-secret",
  "codeChallengeMethod": "S256",
  "cookies": {
    "login": "pylon_login_session",
    "logout": "pylon_logout_session",
  },
  "defaults": {
    "acrValues": null,
    "audience": null,
    "maxAge": null,
    "prompt": null,
    "responseType": "code",
    "scope": [
      "openid",
      "offline_access",
      "email",
      "profile",
    ],
  },
  "dynamicRedirectDomains": [],
  "errorRedirect": "/auth/error",
  "expose": {
    "accessToken": false,
    "idToken": false,
    "scope": false,
    "subject": false,
  },
  "issuer": "https://issuer.com",
  "pathPrefix": "/auth",
  "refresh": {
    "maxAge": "1h",
    "mode": "half_life",
  },
  "staticRedirect": {
    "login": null,
    "logout": null,
  },
  "tokenExpiry": "1d",
}
`;

exports[`parseAuthConfig should merge defaults with options 1`] = `
{
  "clientId": "test-client-id",
  "clientSecret": "test-client-secret",
  "codeChallengeMethod": "plain",
  "cookies": {
    "login": "pylon_login_session",
    "logout": "pylon_logout_session",
  },
  "defaults": {
    "acrValues": null,
    "audience": null,
    "maxAge": null,
    "prompt": null,
    "responseType": "code",
    "scope": [
      "openid",
      "offline_access",
      "email",
      "profile",
    ],
  },
  "dynamicRedirectDomains": [
    "https://client.com",
  ],
  "errorRedirect": "/auth/error",
  "expose": {
    "accessToken": false,
    "idToken": false,
    "scope": false,
    "subject": false,
  },
  "issuer": "https://issuer.com",
  "pathPrefix": "/auth",
  "refresh": {
    "maxAge": "6m",
    "mode": "none",
  },
  "staticRedirect": {
    "login": "https://client.com/login/static",
    "logout": null,
  },
  "tokenExpiry": "1d",
}
`;
